util/options.hh

```diff
@@ -690,7 +690,31 @@ struct subset_options_t : option_group_t
 
   void add_options (option_parser_t *parser) override;
 
+  hb_bool_t* bool_for(hb_subset_flags_t flag)
+  {
+    for (unsigned i = 0; i < sizeof(int) * 8; i++)
+    {
+      if (1 << i == flag)
+        return &flags[i];
+    }
+    return &flags[sizeof(int) * 8 - 1];
+  }
+
   unsigned num_iterations;
+
+  hb_subset_input_t * get_input ()
+  {
+    for (unsigned i = 0; i < sizeof(int) * 8; i++)
+    {
+      hb_subset_input_set_flags (input,
+                                 (hb_subset_flags_t) (1 << i),
+                                 flags[i] ? HB_SUBSET_FLAGS_ALL : HB_SUBSET_FLAGS_NONE);
+    }
+    return input;
+  }
+
+  hb_bool_t flags[sizeof(int) * 8] = {0};
+
```

Comment:
So, instead of this `bool_for` and `get_input`, I suggest you use `G_OPTION_ARG_CALLBACK`.